app/src/main/java/com/android/streetworkapp/model/event/EventRepositoryFirestore.kt

```diff
@@ -66,4 +67,21 @@ class EventRepositoryFirestore(private val db: FirebaseFirestore) : EventReposit
       Log.e("FirestoreError", "Error creating event: ${e.message}")
     }
   }
+
+  fun documentToEvent(document: DocumentSnapshot): Event? {
+    return if (document.exists()) {
+      Event(
+          eid = document.id,
+          title = document["title"] as String,
+          description = document["description"] as String,
+          participants = (document["participants"] as Long).toInt(),
+          maxParticipants = (document["maxParticipants"] as Long).toInt(),
+          date = document["date"] as Timestamp,
+          owner = document["owner"] as String,
+          listParticipants = document["listParticipants"] as List<String>,
+          parkId = document["parkId"] as String)
+    } else {
+      null
```

Comment:
Suggestion: would be nice to add a `Log.e` in order to track some unexpected behaviours

Reply:
I agree, good call!